app/main/views/suppliers.py

```diff
@@ -605,60 +603,49 @@ def company_details():
                 extra={
                     'duns': session.get('duns_number'),
                     'form_errors': ",".join(chain.from_iterable(form.errors.values()))})
-            return render_template(
-                "suppliers/create_company_details.html",
-                form=form,
-                errors=[{'question': form[key].label, 'input_name': form[key].name} for key in form.errors]
-            ), 400
 
-    if form.company_name.name in session:
-        form.company_name.data = session[form.company_name.name]
+    else:
+        if form.company_name.name in session:
+            form.company_name.data = session[form.company_name.name]
 
-    if form.contact_name.name in session:
-        form.contact_name.data = session[form.contact_name.name]
+        if form.contact_name.name in session:
+            form.contact_name.data = session[form.contact_name.name]
 
-    if form.email_address.name in session:
-        form.email_address.data = session[form.email_address.name]
+        if form.email_address.name in session:
+            form.email_address.data = session[form.email_address.name]
 
-    if form.phone_number.name in session:
-        form.phone_number.data = session[form.phone_number.name]
+        if form.phone_number.name in session:
+            form.phone_number.data = session[form.phone_number.name]
+
+    errors = get_errors_from_wtform(form)
 
     return render_template(
         "suppliers/create_company_details.html",
         form=form,
-    ), 200
+        errors=errors,
+    ), 200 if not errors else 400
 
 
-@main.route('/create/account', methods=['GET'])
+@main.route('/create/account', methods=['GET', 'POST'])
 def create_your_account():
-    current_app.logger.info(
-        "suppliercreate: get create-your-account supplier_id:{}".format(
-            session.get('email_supplier_id', 'unknown')))
-    form = EmailAddressForm()
-
-    return render_template(
-        "suppliers/create_your_account.html",
-        form=form,
-        email_address=session.get('account_email_address', '')
-    ), 200
-
+    current_app.logger.info("suppliercreate: {} create-your-account supplier_id:{}".format(
+        request.method,
+        session.get('email_supplier_id', 'unknown'))
+    )
```

Comment:
The indenting here feels a bit odd?

Reply:
I think it's the combination of hanging and inline closing braces which is a little confusing...

Reply:
To be fair though it's not the only place in this app where I've seen that